src/main/resources/static/javascript/mainApp/timeout.controller.js

```diff
@@ -1,195 +1,117 @@
             });
         });
 
         /**
-         * Clear timeouts and intervals when DOM is destroyed.
+         *  Checks on time of inactivity, if time is meet, log out user.
          */
-        $scope.$on("$destroy", function (event) {
-            if (angular.isDefined(createTimeoutModalPromise)) {
-                $timeout.cancel(createTimeoutModalPromise);
-                createTimeoutModalPromise = {};
+        $scope.timerIncrement = function () {
+            $scope.idleTime++;
+            if ($scope.idleTime === 25) {// Create warning modal when 5 min left
+                $scope.countdownTimer = setInterval(timer, 1000);
+                $scope.createTimeoutModal();
             }
-            if (angular.isDefined(countdownTimerPromise)) {
-                $timeout.cancel(countdownTimerPromise);
-                countdownTimerPromise = {};
+            if ($scope.idleTime === 30) { // Logout user after 30 min has passed
+                let r = new XMLHttpRequest();
+                r.open("POST", "/uhgroupings/logout", false);
+                r.setRequestHeader("X-XSRF-TOKEN", $scope.getCookie("XSRF-TOKEN"));
+                r.send();
+                $window.location.href = "/uhgroupings/";
             }
-        });
+        };
 
         /**
-         * Create a countdown timer.
+         * Creates a countdown timer.
          */
         function timer() {
-            $scope.timeRemaining = secondsToMinutes($scope.secondsRemaining);
-            if ($scope.secondsRemaining <= 0) {
-                $scope.logoutOnIdle();
-                restartTimeouts();
-            }
-            $scope.secondsRemaining--;
-        }
-
-        /**
-         * Convert seconds to minutes and return as a formatted string.
-         */
-        function secondsToMinutes(seconds) {
-            let minutes = Math.round((seconds - 30) / 60);
-            let remainingSeconds = seconds % 60;
+            let minutes = Math.round(($scope.seconds - 30) / 60);
+            let remainingSeconds = $scope.seconds % 60;
             if (remainingSeconds < 10) {
                 remainingSeconds = "0" + remainingSeconds;
             }
-            return `${minutes}:${remainingSeconds}`;
-        }
-
-        /**
-         * Restart createTimeoutModalPromise.
-         */
-        function restartTimeouts() {
-            if (angular.isDefined(createTimeoutModalPromise)) {
-                createTimeoutModalPromise = {};
-                createTimeoutModalPromise = $timeout(() => {
-                    $scope.createTimeoutModal();
-                }, MAX_TIME_IDLE);
-
-                createTimeoutModalPromise.then(() => {
-                    // timeout ends and modal is created
-                }, () => {
-                    // User resets timer or function execution fails
-                    restartTimeouts();
-                });
+            document.getElementById("countdown").innerHTML = minutes + ":" + remainingSeconds;
```

Comment:
Can we avoid using document.getElementById? I believe `$scope.timeLeft = minutes + ":" + remainingSeconds;` should accomplish the same thing.